trimming: improve output on failure

topolarity opened the pull request:
Remove the unnecessary stack trace and print the error/warning summary on a single line.

Before:
```console
...
Trim verify finished with 1 warning.

Trim verify finished with 57 errors.

fatal: error thrown and no exception handler available.
ErrorException("verify_typeinf_trim failed")
error at ./error.jl:44
verify_typeinf_trim at ./../usr/share/julia/Compiler/src/verifytrim.jl:341
unknown function (ip: 0x7314d73ccaaa) at (unknown file)
jl_apply at /home/user/jc/jh-juliac-mwe/julia/src/julia.h:2320 [inlined]
jl_f__call_latest at /home/user/jc/jh-juliac-mwe/julia/src/builtins.c:883
#invokelatest#1 at ./essentials.jl:1090 [inlined]
invokelatest at ./essentials.jl:1086 [inlined]
verify_typeinf_trim at ./../usr/share/julia/Compiler/src/typeinfer.jl:1371 [inlined]
typeinf_ext_toplevel at ./../usr/share/julia/Compiler/src/typeinfer.jl:1365
jfptr_typeinf_ext_toplevel_124479 at /home/user/jc/jh-juliac-mwe/julia/usr/lib/julia/sys.so (unknown line)
jl_apply at /home/user/jc/jh-juliac-mwe/julia/src/julia.h:2320 [inlined]
jl_create_native_impl at /home/user/jc/jh-juliac-mwe/julia/src/aotcompile.cpp:667
jl_precompile_trimmed at /home/user/jc/jh-juliac-mwe/julia/src/precompile_utils.c:385 [inlined]
ijl_create_system_image at /home/user/jc/jh-juliac-mwe/julia/src/staticdata.c:3419
ijl_write_compiler_output at /home/user/jc/jh-juliac-mwe/julia/src/precompile.c:155
ijl_atexit_hook at /home/user/jc/jh-juliac-mwe/julia/src/init.c:278
jl_repl_entrypoint at /home/user/jc/jh-juliac-mwe/julia/src/jlapi.c:1125
main at /home/user/jc/jh-juliac-mwe/julia/cli/loader_exe.c:58
unknown function (ip: 0x7314e1029d8f) at /lib/x86_64-linux-gnu/libc.so.6
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
_start at /home/user/jc/jh-juliac-mwe/julia/usr/bin/julia (unknown line)

Failed to compile ./test/trimming/hello.jl
```

After:
```console
...
Trim verify finished with 57 errors, 1 warning.

Failed to compile ./test/trimming/hello.jl
```

topolarity committed:
trimming: improve output on failure

Remove the unnecessary stack trace and print the error/warning summary
on a single line.

JeffBezanson commented on src/precompile_utils.c:
```diff
@@ -382,7 +382,14 @@ static void *jl_precompile_trimmed(size_t world)
+    JL_TRY {
+        native_code = jl_create_native(m, NULL, jl_options.trim, 0, world);
+    } JL_CATCH {
+        // The verification check (presumably) failed. The error message should
```
We should do something to make sure it was actually the verification failing that caused the exception.

topolarity committed:
Add `TrimFailure` error type

And use it to confirm that the error we got from inference is actually a
trim verification failure.
